src/sync/sync_user.cpp

```diff
@@ -32,6 +35,9 @@ SyncUser::SyncUser(std::string refresh_token, std::string identity,
 , m_refresh_token(std::move(refresh_token))
 , m_identity(std::move(identity))
 {
+    if (s_binding_context_factory) {
```

Comment:
You'd need to hold the mutex when checking / accessing this value, otherwise there's no synchronization between these accesses and `set_binding_context_factory`.